File: idea-plugin/src/main/java/com/lingce/aicleaner/FileCategory.java
package com.lingce.aicleaner;

enum FileCategory {
    TEMP("Temporary file"),
    AI_GENERATED_USELESS("AI-generated throwaway file"),
    PROJECT_CONFIG("Project configuration file"),
    AI_CONFIG("AI assistant configuration file"),
    SUSPICIOUS("Suspicious or unknown file"),
    NORMAL("Normal project file"),
    UNKNOWN("Unknown");

    private final String label;

    FileCategory(String label) {
        this.label = label;
    }

    String getLabel() {
        return label;
    }
}

File: idea-clean-guard/src/main/kotlin/com/lingce/cleanguard/model/FileCategory.kt
package com.lingce.cleanguard.model

enum class FileCategory(val displayName: String, val autoAction: AutoAction) {
    TMP("临时文件", AutoAction.DELETE),
    AI_GENERATED_USELESS("AI 生成（无用）", AutoAction.DELETE),
    AI_CONFIG("AI 配置文件", AutoAction.EXCLUDE),
    PROJECT_CONFIG("项目配置文件", AutoAction.EXCLUDE),
    SUSPICIOUS("可疑文件", AutoAction.NONE),
    SAFE("安全", AutoAction.NONE),
    UNKNOWN("未分类", AutoAction.NONE);

    enum class AutoAction {
        DELETE, EXCLUDE, NONE
    }
}

data class ClassifiedFile(
    val path: String,
    val category: FileCategory,
    val reason: String,
    val confidence: Double = 1.0,
)
